File: Back-End/LeilaoEletronico/Domain/Models/Entities/Product.cs
using Domain.Models;
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Product
    {
        public int Id { get;  set; }
        public string ProductName { get;  set; }
        public string ProductDescription { get;  set; }
        public decimal InitialOffer { get; set; }
        public DateTime BidsClosingDate { get; set; }
        public IEnumerable<Offer> Bids { get; set; }
    }
}

File: Front-End/Leilao-Eletronico/src/app/Models/product.ts
import { Offer } from "./Offer";

export interface Product {
  id?: number;
  responsibleName: string;
  productName: string;
  productDescription: string;
  initialOffer: number;
  bidsClosingDate: Date | string;
  bids: [Offer] | [];
  lastBidValue: number;
  lastBidDate: Date | string;
  lastBidResponsible: string;
}
